apps/web/pages/api/user/[id].ts

```diff
@@ -11,12 +12,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     return res.status(401).json({ message: "Not authenticated" });
   }
 
+  const stringToNumber = z.string().transform((val) => parseInt(val));
+
   const userIdQuery = req.query?.id ?? null;
   const userId = Array.isArray(userIdQuery)
     ? parseInt(userIdQuery.pop() || "")
-    : typeof userIdQuery === "string"
-    ? parseInt(userIdQuery)
-    : userIdQuery;
+    : stringToNumber.parse(req.query?.id);
```

Comment:
Since we're not using defaultHandler this error won't be matched so we can:
1.  Refactor this endpoint to use `defaultHandler` and `defaultResponder` or...
2. use `safeParse` and check for success.